### DISCUSSION TITLE
Working example

### DISCUSSION THREAD

**Discussion by jamesg1** [aem-vite/vite-aem-plugin#4] [General]
Hi, firstly love this concept of this plugin. I am trying to use this to improve dev exp with vite with AEM. I've ran into some issues getting the plugin to work. Is there any chance you can provide a working example solution?

I'm getting errors such as `Error: No input option(s) was provided via rollupOptions.input.`

I had to also upgrade vite to 2.9.15 in order to get the Types on the plugin to work.

I have an existing vite project with some config in order to get vite working for aem.

```
 {
            ...aemViteImportRewriter({
              publicPath: aemClientlibsPath,
            }),
            apply: 'build',
            enforce: 'pre',
          },

**Comment by jamesg1** [comment]
Hey @cshawaus any updates on this one, would love to get your plugin working properly.

**Reply by cshawaus** [reply]
Hi @jamesg1,

I'm sorry for the extended delay in replying. Can you provide a full version of your Vite config as the error suggests you may be using an unsupported input format?

Refer to https://www.aemvite.dev/guide/front-end/vite/#source-structure for an example of what this needs to look like.

Cheers

**Reply by maciej-panecki-wttech** [reply]
While it is nice to try to solve specific issue with one person project it would be beneficial if you would create a working test project so that people could see how it should be set up. I'm also struggling with some of the steps that are mentioned but not documented. For example `Tools such as Adobe's aem-clientlib-generator will work perfectly fine with AEM Vite as it is executed after a build.` are not configured to be triggered anywhere after the build and need to be run manually
